### ISSUE TITLE
initial GET not populating Form Elements? 

### ISSUE THREAD

**Issue by timryder** [VolkovLabs/volkovlabs-form-panel#155]
I have successfully made use of the POST functionality of DMP, it is nicely entering values into my PosgreSQL database from Granfa no issues. However when I created an endpoint for getting the initial value to populate my form elements with, it isn't doing that. First off it is not inherently obvious how this plugin works, but I am assuming that similar to how the POST generates a JSON using the Form Element's ID as the JSON variable name and then the elements value gets composed into the JSON value.... i assume the GET will do the same but in reverse. Use the JSON variable name as a lookup for the Form Element ID and if found, populate it's value with the JSON value from the GET body. 

Well I am using the browser debugging tools and I am successfully seeing the GET response 200, with my requested data in a JSON format, but it isn't populating the form elements with those values? Any idea what is wrong? 

![image](https://user-images.githubusercontent.com/237432/231878377-1e9a5634-0f91-4102-84b9-4857d168d152.png)

![image](https://user-images.githubusercontent.com/237432/231878492-2e5ecb60-837e-40eb-874f-7851a07ac86e.png)

![image](https://user-images.githubusercontent.com/237432/231878525-1a5a8ea5-4b56-4a5d-80a5-e2a4a0a10603.png)

![image](https://user-images.githubusercontent.com/237432/231878809-3d92049c-0719-4dc0-93cd-cf7e8528d011.png)

**Comment by timryder** [comment]
I realized my problem, I was returning the JSON object as the value of another JSON object.... sorry for the waste of a report.

**Comment by mikhail-vl** [comment]
@timryder Thank you for providing the solution. You are correct. JSON fields should be at the root of the payload.

  Regarding `First off, it is not inherently obvious how this plugin works`, how would you improve it? We provided documentation https://volkovlabs.io/plugins/volkovlabs-form-panel/ and YouTube tutorials, have you seen them?

  Let us know what we need to improve and make it effortless for the community.

**Comment by timryder** [comment]
Sure happy to comment!

Nowhere in the documentation does it describe anywhere how the GET method works. There are some for POST only.  I found by reading a particular thread here in the issues how the Initial GET works. You replied to another user that the plugin parses the return JSON object automatically and thus consumes it.

He was asking about how to re-evaluate the JSON in the payload but your reply actually finally explained how the plugin handles the data and tries to link the variable names to Form Element IDs if they exist.  Then applies the values of the variables to the Form Element values.  

This description is nowhere in the documentation. 

Also for the same the POST, I saw examples yes, but no description of the process. I only learned what was happening in the post by looking at the browser debugger Network tab. I think the process for all of the functions in this plugin need to be better described. The fact that when you click the Upload button it Generates a JSON by combining the Form Element ID's with their values isn't obvious.  

Let me know if you want me to add context to your descriptions I can help!
